core/src/main/java/org/sonarsource/sonarlint/core/VersionSoonUnsupportedHelper.java

```diff
@@ -0,0 +1,113 @@
+import org.sonarsource.sonarlint.core.clientapi.client.message.MessageType;
+import org.sonarsource.sonarlint.core.clientapi.client.message.ShowSoonUnsupportedMessageParams;
+import org.sonarsource.sonarlint.core.commons.ConnectionKind;
+import org.sonarsource.sonarlint.core.commons.Version;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+import org.sonarsource.sonarlint.core.event.BindingConfigChangedEvent;
+import org.sonarsource.sonarlint.core.event.ConfigurationScopesAddedEvent;
+import org.sonarsource.sonarlint.core.repository.config.ConfigurationRepository;
+import org.sonarsource.sonarlint.core.repository.connection.ConnectionConfigurationRepository;
+import org.sonarsource.sonarlint.core.serverapi.system.ServerInfo;
+import org.sonarsource.sonarlint.core.serverconnection.VersionUtils;
+
+@Named
+@Singleton
+public class VersionSoonUnsupportedHelper {
+
+  private static final String UNSUPPORTED_NOTIFICATION_ID = "sonarlint.unsupported.%s.id";
+  private static final String NOTIFICATION_MESSAGE = "The version used by the current connection '%s' will be soon unsupported. " +
+    "Please consider upgrading to the latest %s LTS version to ensure continued support and access to the latest features.";
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+  private final SonarLintClient client;
+  private final ConfigurationRepository configRepository;
+  private final ConnectionConfigurationRepository connectionRepository;
+  private final ServerApiProvider serverApiProvider;
+
+  public VersionSoonUnsupportedHelper(SonarLintClient client, ConfigurationRepository configRepository,
+    ConnectionConfigurationRepository connectionRepository, ServerApiProvider serverApiProvider) {
+    this.client = client;
+    this.configRepository = configRepository;
+    this.connectionRepository = connectionRepository;
+    this.serverApiProvider = serverApiProvider;
+  }
+
+  @Subscribe
+  public void configurationScopesAdded(ConfigurationScopesAddedEvent event) {
+    var configScopeIds = event.getAddedConfigurationScopeIds();
+    notifyIfSoonUnsupported(configScopeIds);
+  }
+
+  @Subscribe
+  public void bindingConfigChanged(BindingConfigChangedEvent event) {
+    var configScopeId = event.getConfigScopeId();
+    var connectionId = event.getNewConfig().getConnectionId();
+    if (connectionId != null) {
+      notifyIfSoonUnsupported(configScopeId, connectionId);
+    }
+  }
+
+  private void notifyIfSoonUnsupported(Set<String> configScopeIds) {
+    configScopeIds.forEach(configScopeId -> {
+      var effectiveBinding = configRepository.getEffectiveBinding(configScopeId);
+      if (effectiveBinding.isPresent()) {
+        var connectionId = effectiveBinding.get().getConnectionId();
+        notifyIfSoonUnsupported(configScopeId, connectionId);
+      }
+    });
+  }
+
+  private void notifyIfSoonUnsupported(String configScopeId, String connectionId) {
```

Comment:
At this point, we don't know if a notification will be sent, so I would rename the method to something like: `checkIfSoonUnsupported`